libraries/AP_UAVCAN/AP_UAVCAN_Servers.cpp

```diff
@@ -110,6 +113,168 @@ class AP_UAVCAN_RestartRequestHandler : public uavcan::IRestartRequestHandler {
+            return;
+        }
+
+        _queued_parameter->copy_name_token(_queued_parameter_token,
+                                           (char*)last_param_name,
+                                           AP_MAX_NAME_SIZE,
+                                           true);
+        out_name = last_param_name; // no, really, we can do this....
+    };
+
+    /**
+     * Assign by name if exists.
+     */
+    void assignParamValue(const uavcan::IParamManager::Name& name,
+                          const uavcan::IParamManager::Value& value) override {
+        // gcs().send_text(MAV_SEVERITY_INFO, "Assign: %s", name.c_str());
+        ap_var_type p_type;
+        AP_Param *p = get_param_by_name(name.c_str(), &p_type);
+        if (p == nullptr) {
+            return;
+        }
+
+        float set_value;
+        const float *float_value = value.as<uavcan::protocol::param::Value::Tag::real_value>();
+        if (float_value != nullptr) {
+            set_value = *float_value;
+        } else {
+            auto *y = value.as<uavcan::protocol::param::Value::Tag::integer_value>();
+            if (y == nullptr) {
+                return;
+            }
+            set_value = *y;
+        }
+
+        p->user_set_value(set_value, p_type);
+
+        AP::logger().Write_Parameter(name.c_str(), p->cast_to_float(p_type));
+    }
+
+    /**
+     * Read by name if exists, otherwise do nothing.
+     */
+    void readParamValue(const uavcan::IParamManager::Name& name,
+                        uavcan::IParamManager::Value& out_value) const override {
+        // gcs().send_text(MAV_SEVERITY_INFO, "Read: %s", name.c_str());
+        ap_var_type p_type;
+        const AP_Param *p = const_cast<AP_UAVCAN_ParamManager*>(this)->get_param_by_name(name.c_str(), &p_type);
+        if (p == nullptr) {
+            return;
+        }
+        switch (p_type) {
+        case AP_PARAM_INT8:
+            out_value.to<uavcan::protocol::param::Value::Tag::integer_value>() = p->as_int8_t(p_type);
+            break;
+        case AP_PARAM_INT16:
+            out_value.to<uavcan::protocol::param::Value::Tag::integer_value>() = p->as_int16_t(p_type);
+            break;
+        case AP_PARAM_INT32:
+            out_value.to<uavcan::protocol::param::Value::Tag::integer_value>() = p->as_int32_t(p_type);
+            break;
+        case AP_PARAM_FLOAT:
+            out_value.to<uavcan::protocol::param::Value::Tag::real_value>() = p->cast_to_float(p_type);
+            break;
+        case AP_PARAM_VECTOR3F:
+        case AP_PARAM_GROUP:
+        case AP_PARAM_NONE:
+            gcs().send_text(MAV_SEVERITY_INFO, "Unknown param type");
+            break;
+        }
+    }
+
+    /**
+     * Save all params to non-volatile storage.
+     * @return Negative if failed.
+     */
+    int saveAllParams() override {
+        gcs().send_text(MAV_SEVERITY_INFO, "saveAll does nothing");
+        return -1;
```

Comment:
I would think this one should actually succeed since parameters are already saved. From a UAVCAN point of view, I think this is a failure and the other node would expect the parameters aren't saved since this failed...

Reply:
Change to return success.